js/ui/viewSelector.js

```diff
@@ -141,8 +141,8 @@ const ShowOverviewAction = new Lang.Class({
 const ViewsDisplayLayout = new Lang.Class({
     Name: 'ViewsDisplayLayout',
     Extends: Clutter.BinLayout,
-    Signals: { 'allocated-size-changed': { param_types: [GObject.TYPE_INT,
-                                                         GObject.TYPE_INT] } },
+    Signals: { 'grid-available-size-changed': { param_types: [GObject.TYPE_INT,
+                                                              GObject.TYPE_INT] } },
```

Comment:
I understand the name change and I agree it makes sense as we're now sending the available size that we calculated for the grid (instead of the size allocated to the ViewsDisplayContainer), but I thinking twice about this I don't think we should emit any signal at all:

This signal is being connected to from ViewsDisplayContainer only, and its only purpose is to make sure we call `adaptToSize()` over the grid container before the allocation, which we could do directly from here by calling `gridContainerActor.adaptToSize()`, allowing us to remove all the code related to this signal from here and move the logic of `_onGridAvailableSizeChanged()` from ViewsDisplayContainer right into a method in the layout manager.

It would be nice if you could make that change, I think it would simplify things and make it more readable.